scipy/fft/_pocketfft/tests/test_real_transforms.py

```diff
@@ -194,8 +202,17 @@ def test_complex(transform, dtype):
     assert_array_almost_equal(x, y)
 
 
+if TYPE_CHECKING:
+    Numbers = Union[Type[np.double], Type[np.float32], Type[int]]
+    DecMapType = Dict[
+        Tuple[
+            partial, Numbers, int
+        ],
+        int
+    ]
```

Comment:
```suggestion
DecMapType = Dict[Tuple[Callable, type, int], int]
```

Reply:
Don't forget the subscription:
```suggestion
    DecMapType = Dict[
        # Or just `Type[np.floating]` instead of the union
        Tuple[Callable[..., np.ndarray], Union[Type[np.float32], Type[np.float64]], int],  
        int,
    ]
```

Reply:
I just tested with mypy and `Callable` with no arguments works exactly as expected. 

`Union[Type[np.float32], Type[np.float64]]` is also wrong because we have tests for `np.longdouble` as well.

Reply:
> I just tested with mypy and Callable with no arguments works exactly as expected.

I'ts no suprise that that `Callable` without arguments works as expected, as it is equivalent to `Callable[..., Any]`.

> `Union[Type[np.float32], Type[np.float64]]` is also wrong because we have tests for `np.longdouble` as well.

In that case `Type[np.floating]` would definitely be more appropriate.